cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -401,6 +418,45 @@ canonDbPath NodeConfiguration{ncDatabaseFile = DbFile dbFp} = do
   createDirectoryIfMissing True fp
   return fp
 
+#ifdef UNIX
+-- | Make sure the VRF private key file is readable only
+-- by the current process owner the node is running under.
+checkVRFFilePermissions :: Maybe FilePath -> ExceptT VRFPrivateKeyFilePermissionError IO ()
+checkVRFFilePermissions mFp =
+  case mFp of
+    Just vrfPrivKey -> do
+      fs <- liftIO $ getFileStatus vrfPrivKey
+      let fm = fileMode fs
+
+      -- Check the the VRF private key file does not give read/write/exec permissions to others.
+      when (hasOtherPermissions fm)
+           (left $ OtherPermissionsExist vrfPrivKey)
+      -- Check the the VRF private key file does not give read/write/exec permissions to any group.
+      when (hasGroupPermissions fm)
+           (left $ GroupPermissionsExist vrfPrivKey)
+      -- Check that the VRF private key file has owner read permissions.
+      unless (hasOwnerPermissions fm)
+             (left $ OwnerDoesNotHaveReadWrite vrfPrivKey)
```

Comment:
This check is unnecessary since we will read the file anyway. We only need to check the negative permissions not the positive ones.